intelmq/bots/outputs/elastic/output.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+import sys
+from json import loads
+from collections.abc import Mapping
+
+from elasticsearch import Elasticsearch
+
+from intelmq.lib.bot import Bot
+
+#
+# TODO: recursion depth check
+
+
+def replace_keys(obj, key_char='.', replacement='_'):
+    if isinstance(obj, Mapping):
+        replacement_obj = {}
+        for key, val in obj.items():
+            replacement_key = key.replace(key_char, replacement)
+            replacement_obj[replacement_key] = replace_keys(val, key_char, replacement)
+        return replacement_obj
+    return obj
+
+
+class ElasticsearchOutputBot(Bot):
+
+    def init(self):
+        self.elastic_host = getattr(self.parameters,
+                                    'elastic_host', '127.0.0.1')
+        self.elastic_port = getattr(self.parameters,
+                                    'elastic_port', '9200')
+        self.elastic_index = getattr(self.parameters,
+                                     'elastic_index', 'intelmq')
+        self.elastic_doctype = getattr(self.parameters,
+                                       'elastic_doctype', 'events')
+        self.sanitize_keys = getattr(self.parameters,
+                                     'sanitize_keys', True)
+        self.replacement_char = getattr(self.parameters,
+                                        'replacement_char', '_')
+        self.flatten_fields = getattr(self.parameters,
+                                      'flatten_fields', [])
+        if isinstance(self.flatten_fields, str):
+            self.flatten_fields = self.flatten_fields.split(',')
+
+        self.es = Elasticsearch([{'host': self.elastic_host,
+                                  'port': self.elastic_port}])
+        if not self.es.indices.exists(self.elastic_index):
+            self.es.indices.create(index=self.elastic_index, ignore=400)
+
+    def process(self):
+        event = self.receive_message()
+        event_dict = event.to_dict(hierarchical=False)
+
+        for field in self.flatten_fields:
+            if field in event_dict:
+                val = event_dict[field]
+                # if it string try to convert to json
```

Comment:
@navtej let me ask, when you mean "flatten field"  do you mean to also "flatten" the "." (dot) away ?
I remember @SYNchroACK having done that in early days. Back then we used logstash to still re-write things.

Anyway, I'd like to know (maybe we can get some feedback from multiple users of ES on the mailing list) what format per se would be the most suitable for ES?